File: generated-sources/c/mojang-sessions/api/LoginAPI.c
#include <stdlib.h>
#include <stdio.h>
#include <ctype.h>
#include "apiClient.h"
#include "cJSON.h"
#include "keyValuePair.h"
#include "player_profile.h"
#include "session_api_error.h"

#define MAX_BUFFER_LENGTH 4096
#define intToStr(dst, src) \
    do {\
    char dst[256];\
    snprintf(dst, 256, "%ld", (long int)(src));\
}while(0)

// Gets the player's game profile
//
// This will return the player's username plus any additional information about them (e.g. skins). This has a much stricter rate limit: You can request the same profile once per minute, however you can send as many unique requests as you like.
//
player_profile_t*
LoginAPI_getPlayerProfile(apiClient_t *apiClient, char* stripped_uuid, int unsigned) {
    list_t    *localVarQueryParameters = list_create();
    list_t    *localVarHeaderParameters = NULL;
    list_t    *localVarFormParameters = NULL;
    list_t *localVarHeaderType = list_create();
    list_t *localVarContentType = NULL;
    char      *localVarBodyParameters = NULL;

    // create the path
    long sizeOfPath = strlen("/session/minecraft/profile/{stripped_uuid}")+1;
    char *localVarPath = malloc(sizeOfPath);
    snprintf(localVarPath, sizeOfPath, "/session/minecraft/profile/{stripped_uuid}");


    // Path Params
    long sizeOfPathParams_stripped_uuid = strlen(stripped_uuid)+3 + strlen("{ stripped_uuid }");

    if(stripped_uuid == NULL) {
        goto end;
    }
    char* localVarToReplace_stripped_uuid = malloc(sizeOfPathParams_stripped_uuid);
    sprintf(localVarToReplace_stripped_uuid, "%s%s%s", "{", "stripped_uuid", "}");

    localVarPath = strReplace(localVarPath, localVarToReplace_stripped_uuid, stripped_uuid);

    // query parameters
    char *keyQuery_unsigned;
    int valueQuery_unsigned;
    keyValuePair_t *keyPairQuery_unsigned = 0;
    if (unsigned)
    {
    //not string
    keyQuery_unsigned = strdup("unsigned");
    valueQuery_unsigned = unsigned;
    keyPairQuery_unsigned = keyValuePair_create(keyQuery_unsigned, &valueQuery_unsigned);
    list_addElement(localVarQueryParameters,keyPairQuery_unsigned);
    }
    list_addElement(localVarHeaderType,"application/json"); //produces
    apiClient_invoke(apiClient,
                    localVarPath,
                    localVarQueryParameters,
                    localVarHeaderParameters,
                    localVarFormParameters,
                    localVarHeaderType,
                    localVarContentType,
                    localVarBodyParameters,
                    "GET");

    if (apiClient->response_code == 200) {
        printf("%s\n","The player profile and all properties");
    }
    if (apiClient->response_code == 400) {
        printf("%s\n","UUID is invalid");
    }
    //nonprimitive not container
    player_profile_t *elementToReturn = player_profile_parseFromJSON(apiClient->dataReceived);
    if(elementToReturn == NULL) {
        // return 0;
    }

    //return type
    apiClient_free(apiClient);
    list_free(localVarQueryParameters);
    
    
    list_free(localVarHeaderType);
    
    free(localVarPath);
    free(localVarToReplace_stripped_uuid);
    free(keyQuery_unsigned);
    keyValuePair_free(keyPairQuery_unsigned);
    return elementToReturn;
end:
    return NULL;

}

File: generated-sources/dart/mojang-sessions/lib/api/login_api.dart
part of openapi.api;



class LoginApi {
  final ApiClient apiClient;

  LoginApi([ApiClient apiClient]) : apiClient = apiClient ?? defaultApiClient;

  /// Gets the player&#39;s game profile
  ///
  /// This will return the player&#39;s username plus any additional information about them (e.g. skins). This has a much stricter rate limit: You can request the same profile once per minute, however you can send as many unique requests as you like.
  Future<PlayerProfile> getPlayerProfile(String strippedUuid, { bool unsigned }) async {
    Object postBody;

    // verify required params are set
    if(strippedUuid == null) {
     throw new ApiException(400, "Missing required param: strippedUuid");
    }

    // create path and map variables
    String path = "/session/minecraft/profile/{stripped_uuid}".replaceAll("{format}","json").replaceAll("{" + "stripped_uuid" + "}", strippedUuid.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    if(unsigned != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "unsigned", unsigned));
    }

    List<String> contentTypes = [];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
    }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return apiClient.deserialize(response.body, 'PlayerProfile') as PlayerProfile;
    } else {
      return null;
    }
  }
}
